smbclient/shutil.py

```diff
@@ -0,0 +1,178 @@
+from __future__ import unicode_literals
+
+from ntpath import join, basename, splitdrive
+
+import sys
+import os.path
+from os import error as os_error
+
+from smbclient._os import remove, rmdir, listdir, stat, makedirs, readlink, symlink, scandir, copyfile_server_side
+from smbclient.path import islink, isdir
+from smbprotocol.exceptions import SMBOSError
+
+
+class Error(EnvironmentError):
```

Comment:
This seems quite generic, can we use one of the existing SMB Exceptions we have in `smbprotocol.exceptions`?